spdmlib/src/crypto/x509v3.rs

```diff
@@ -692,13 +862,11 @@ pub fn check_leaf_certificate(cert: &[u8], is_alias_cert_model: bool) -> SpdmRes
     //extensions            EXTENSIONS,
     let (_, extension_data) = check_and_get_extensions(&data[t_walker..])?;
 
-    if is_alias_cert_model
-        && find_target_object_identifiers(extension_data, OID_DMTF_SPDM_HARDWARE_IDENTITY)?
-    {
+    if is_alias_cert_model && find_hardware_object_identifiers_in_extensions(extension_data)? {
         info!("Hardware identity OID shall not be present in alias cert!\n");
         Err(SPDM_STATUS_VERIF_FAIL)
     } else if !is_alias_cert_model
-        && !find_target_object_identifiers(extension_data, OID_DMTF_SPDM_HARDWARE_IDENTITY)?
+        && !find_hardware_object_identifiers_in_extensions(extension_data)?
```

Comment:
Why we change a generic function to a specific function?

Reply:
> Why we change a generic function to a specific function?

Because, it is confusing for a generic function to find a target oid. RFC 5280 says
```
Extension  ::=  SEQUENCE  {
    extnID      OBJECT IDENTIFIER,
    critical    BOOLEAN DEFAULT FALSE,
    extnValue   OCTET STRING
                -- contains the DER encoding of an ASN.1 value
                -- corresponding to the extension type identified
                -- by extnID
        }
```
to find a target oid in extnValue need the context of extension type. So it is better to add some specific items in function name to indicate the context of extension type. In this PR, hardware_object_identifiers indicates this function works under SPDM context, to find a
```
id-DMTF-hardware-identity id-spdm-cert-oid :: = {
    spdmOID { id-DMTF-spdm 2 }
    spdmOIDdefinition ABSENT
}

```
it should be more precise.